apps/web/src/pages/checkout/[packSlug].tsx

```diff
@@ -0,0 +1,184 @@
+import {
+  CheckoutMethod,
+  PackAuction,
+  PackStatus,
+  PackType,
+  PublishedPack,
+} from '@algomart/schemas'
+import { GetServerSideProps } from 'next'
+import { useRouter } from 'next/router'
+import useTranslation from 'next-translate/useTranslation'
+import { useEffect } from 'react'
+import { v4 as uuid } from 'uuid'
+
+import { ApiClient } from '@/clients/api-client'
+import { Analytics } from '@/clients/firebase-analytics'
+import { usePaymentProvider } from '@/contexts/payment-context'
+import { Environment } from '@/environment'
+import DefaultLayout from '@/layouts/default-layout'
+import {
+  getAuthenticatedUser,
+  handleUnauthenticatedRedirect,
+} from '@/services/api/auth-service'
+import CheckoutTemplate from '@/templates/checkout-template'
+import { urls } from '@/utils/urls'
+
+export interface CheckoutPageProps {
+  address?: string
+  auctionPackId: string | null
+  currentBid: number | null
+  release: PublishedPack
+}
+
+export default function Checkout({
+  address,
+  auctionPackId,
+  currentBid,
+  release,
+}: CheckoutPageProps) {
+  const { t } = useTranslation()
+  const { query } = useRouter()
+  const paymentProps = usePaymentProvider({
+    auctionPackId,
+    currentBid,
+    release,
+  })
+  const { setAddress, setMethod } = paymentProps
+
+  // Set the address retrieved in server side props
+  useEffect(() => {
+    if (address) {
+      setAddress(address)
+    }
+  }, [address, setAddress])
+
+  // Set the method to the method listed as a query param:
+  useEffect(() => {
+    if (
+      query.method &&
+      Object.values(CheckoutMethod).includes(query.method as CheckoutMethod)
+    ) {
+      setMethod(query.method as CheckoutMethod)
+    } else {
+      setMethod(null)
+    }
+  }, [query.method, setMethod])
```

Comment:
This changes the method based on the query param, so we can send the user to `/checkout/[packSlug]?method=card` directly, and it will direct them to the card payment flow.